js/controller.js

```diff
@@ -1022,7 +1026,11 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
           this.state.screenToShow = CONSTANTS.SCREEN.PAUSE_SCREEN;
         }
         this.state.playerState = CONSTANTS.STATE.PAUSE;
-        this.renderSkin();
+        this.renderSkin().then(() => {
+          if (this.state.cast.isReceiver) {
+            this.showControlBar();
```

Comment:
@marcosmx 
At first I added show/hide/scheduleHide logic in the `withAutoHide` HOC (with additional subscriptions to the mb play/pause events in the controller) but it caused unexpected behaviour since withAutoHide cancels timers in its `componentWillUnmount`. (e.g. _play_ event handler (in which I’m calling startHideControlBarTimer) would be fired before the withAutoHide’s willUnmount)
So I decided to move this logic here

Reply:
I got your point, the same happens to me and I found a better way to implement this without the need to implement a Promise. 
First of all, We need to use the PlayingScreen and PauseScreen without wrapping them with the HOC (withAutoHide), this will remove that weird behaviour with the render cycle.

Why should we use this screens with no-HOC?
That HOC was created to provide or enhance the inner component with logic to handle the user interaction on actions like "mouse enter", "mouse leave", etc... being said that, on the receiver we don't have any user interaction trough the mouse, the player is handled via the API exposed by the core.

I can give you a show case on how to implement this when you get back from vacations, if you want.

@zelibobla ^^